core/domain/role_services.py

```diff
@@ -131,3 +142,69 @@ def get_all_actions(role):
         role_actions.extend(get_all_actions(parent_role))
 
     return list(set(role_actions))
+
+
+def get_updatable_roles():
+    """Returns roles to which a user's role can be updated by admin interface.
+
+    Returns:
+        dict. A dict of all roles that a user can be given by admin.
+            key -> roles in format as stored in backed.
+            value -> roles in format to show users.
+    """
+    updatable_roles = {}
+    for role in PARENT_ROLES:
+        if role not in [feconf.ROLE_GUEST]:
+            updatable_roles[role] = get_human_readable_role(role)
+    return updatable_roles
+
+
+def get_viewable_roles():
+    """Returns roles by which users can be searched in admin interface.
+
+    Returns:
+        dict. A dict of all roles by which user's can be searched.
+            key -> roles in format as stored in backend.
+            value -> roles in format to show users.
+    """
+    viewable_roles = {}
+    for role in PARENT_ROLES:
+        if role not in [feconf.ROLE_EXPLORATION_EDITOR, feconf.ROLE_GUEST]:
+            viewable_roles[role] = get_human_readable_role(role)
+    return viewable_roles
+
+
+def get_role_graph_data():
+    """Returns dict for displaying roles graph.
+
+    Returns:
+        dict. A dict containing data in following format:
+        {
+            finalStateIds: list(str). Roles which have no child.
```

Comment:
But why? They don't make sense in this context, so I am wondering whether StateGraphLayoutService is incorrectly factored and some stuff in it needs to be extracted into the stateGraphViz service.

Alternatively, one way to look at it is that the stateGraphViz service doesn't really guarantee anything about its graph structure when depending on the state graph layout service. But readOnlyGraphService kind of does. So maybe what you can do is, don't compute these things in the backend, but compute them in the readOnlyGraphService (based on the given list of nodes/edges) and pass the whole package to the StateGraphLayoutService.